python/lsst/dax/obscore/obscore_exporter.py

```diff
@@ -312,6 +320,113 @@ def to_csv(self, output: str) -> None:
+            The resulting matches as a VOTable.
+        """
+        # Read the VOTable schema
+        obscore_defn = ResourcePath("resource://lsst.dax.obscore/configs/obscore_nominal.yaml").read()
+        obscore_data = yaml.safe_load(obscore_defn)
+        schema = FelisSchema.model_validate(obscore_data)
+
+        tables = schema.tables
+        if len(tables) != 1:
+            raise RuntimeError("More than one table defined in ObsCore schema")
+        obscore_columns = {column.name: column for column in tables[0].columns}
+
+        votable = astropy.io.votable.tree.VOTableFile()
+        resource = astropy.io.votable.tree.Resource()
+        votable.resources.append(resource)
+
+        fields = []
+        for arrow_field in self.schema:
+            if arrow_field.name in obscore_columns:
+                ffield = obscore_columns[arrow_field.name]
+                votable_datatype = FelisType.felis_type(ffield.datatype.value).votable_name
+                field = astropy.io.votable.tree.Field(
+                    votable,
+                    name=ffield.name,
+                    datatype=votable_datatype,
+                    arraysize=ffield.votable_arraysize,
+                    unit=ffield.ivoa_unit,
+                    ucd=ffield.ivoa_ucd,
+                )
+                fields.append(field)
+            elif arrow_field.name == "em_filter_name":
+                # Non-standard but part of internal standard schema.
+                field = astropy.io.votable.tree.Field(
+                    votable,
+                    name="em_filter_name",
+                    datatype="char",
+                    arraysize="*",
+                )
+                fields.append(field)
+            else:
+                # This must be a non-standard field. Attempt to add it.
+                kwargs = {}
+                datatype = ""
+                if arrow_field.type.equals(pyarrow.int64()):
+                    datatype = "int"
+                elif arrow_field.type.equals(pyarrow.string()):
+                    datatype = "char"
+                    kwargs["arraysize"] = "*"
+                elif arrow_field.type.equals(pyarrow.float64()):
+                    datatype = "double"
+                else:
+                    raise RuntimeError(f"Could not handle unrecognized column {arrow_field.name}")
+                field = astropy.io.votable.tree.Field(
+                    votable,
+                    name=arrow_field.name,
+                    datatype=datatype,
+                    **kwargs,
+                )
+                fields.append(field)
+
+        table0 = astropy.io.votable.tree.TableElement(votable)
+        resource.tables.append(table0)
+        table0.fields.extend(fields)
+
+        def is_none(v: Any) -> bool:
+            return v is None
+
+        chunks = []
+        n_rows = 0
+        for record_batch in self._make_record_batches(self.config.batch_size):
```

Comment:
I was trying to use the existing infrastructure as much as possible so keeping batching seemed fine. I don't think it's possible to do chunked writes of a VOTable within astropy (and the SIAv2 interface will need the full table) so batching isn't overly helpful. I am hoping people will not be exporting millions of rows in VOTable format and SIAv2 should never be trying to deal with queries that return hundreds of thousands of rows. I should probably implement LIMIT in all these queries as an option since it's part of the SIAv2 standard anyhow.

Reply:
A reasonable LIMIT should probably be OK, you can then set batch size to LIMIT and get everything back as one batch so you don't need to do all that gymnastics with `hstack`.

Reply:
I do support MAXREC now so can in theory set this to MAXREC if known, but if I don't know MAXREC I'm not really sure what a reasonable limit to use since -- should I force a MAXREX of 1 million on the basis that people shouldn't be returning more than that for SIAv2? @gpdf do you have a feel for what a reasonable cap should be given that we are accumulating results in memory in a single VOTable and can't stream the results from the server to keep memory under control.